### ISSUE TITLE
Breaking brackets in strings

### ISSUE THREAD

**Issue by npip99** [prettier/prettier#5067]
**Prettier 1.14.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEATArgQwDYBkIDmAFADpQAEl5ABgCTA7YDK6ARgM5gBOAlgA4we0dgDpsCAjAAWAX3Ls2nXgKFR25bBEyo4qEeQAKXCGDjt2PKAXKYY5ALYR2d+q1tgpTHgC84ciFw6XKLUZACUANwgADQgECrCyKCYXMYA7gYpCOzIIJgAbhA8qDEgrFyYYADWcDBMfJWWBMgwXOhwsZbscFwwRpgE9pjIAGY43bEAVuwAHgBCFdW1TJj2cLiWcKPjHSANwT25bqwAnppQpXy8sADqxdLIABwADLFXTnA3FXy5V2Y9+S2sS4cAAjugeCD+oNhkgxtgJiBuvYeC02rsLFZxABFdAQeDbBG7GCYVh3VAPJAAJlirUwPGwTQAwhB7ENclBoECQOhugAVUk5OE7GQyIA)
```sh
--parser babylon
```

**Input:**
```jsx
dualLog(
    `${somelongobject.length} stuff loaded. Processing at most ${thismany} orders.`
);
```

**Output:**
```jsx
log(
  `${
    somelongobject.length
  } stuff loaded. Processing at most ${thismany} orders.`
);
```

**Expected behavior:**
```jsx
log(
    `${somelongobject.length} stuff loaded. Processing at most ${thismany} orders.`
);

```

The output is quite ugly and difficult to read. I understand this was referenced in https://github.com/prettier/prettier/issues/3368, and it seems to have been fixed for single variables ("thismany" could be "thisxxxxxxxxxxxxxxxxxxxxxxmany", and it wouldn't break)

However, it still shouldn't be breaking in the case I described above. There's not a way for me to manually avoid it from breaking either, other than alias somelongobject.length to something else, which isn't too much of a solution. If someone is throwing complex expressions in there, sure, break it, but simple cases such as a.b.c.d, a[b], a[b].c.d[e], or anything that doesn't involve binary operators like +/-/*, is most naturally left unbroken. Putting complex expressions in template literals so rare as far I understand (At least on what I've seen), that the inconvenience from that would be less than the inconvenience of the difficult to read splitting on something like somelongobject.length (And keeping it together doesn't seem to violate any prettier methodology). Additionally, not splitting enough can be fixed manually, while splitting too much means I'm unable to fix it without adding a script to fix it after every prettier run.

At the minimum, I believe it's fair to describe that given
```jsx
`Here's some blah blah blah blah blah blah blah blah blah blah blah: ${obj.len}`

```
and 
```jsx
`Here's some blah blah blah blah blah blah blah blah blah blah blah: ${hereIsSomeRediculouslyLongNameThatIsStillNotSplit}`

```
splitting the former, and not splitting the latter, is illogical. If the former cannot be split, than neither should the latter, so the case described above I think counts as some type of bug (That in my belief would be best fixed by having neither split, and only having variables split when binary operators are used - or, for ease, just never string literals forcibly split).

**Comment by CoolGoose** [comment]
It also happens for objects

```js
{
  text: `Removing environment <strong>${this.data.env_name}</strong> can't be undone.`,
}
```

```js
{
  text: `Removing environment <strong>${
    this.data.env_name
  }</strong> can't be undone.`,
}

**Comment by ValeryVS** [comment]
ESLint in it actual rules don't argue about template strings, and how long they are.
I think Prettier shouldn't try to fit them in "max string length".

**Comment by thorn0** [comment]
fixed long time ago by #5979
